pkg/pool-weighted/contracts/BaseWeightedPool.sol

```diff
@@ -209,14 +217,24 @@ abstract contract BaseWeightedPool is BaseMinimalSwapInfoPool {
         uint256[] memory normalizedWeights = _getNormalizedWeights();
 
         _beforeJoinExit(balances, normalizedWeights);
+
+        uint256 preJoinExitSupply = totalSupply();
```

Comment:
You don't actually need to read it so early, as we haven't yet minted anything.

Also, we might save gas by passing this into `_doJoin`.